### ISSUE TITLE
Use GitHub Actions to generate official Transmission releases

### ISSUE THREAD

**Issue by dechamps** [transmission/transmission#7090]
### What is the issue?

#6873 was caused by what looks like human error when preparing an official Transmission release.

As I suggested in https://github.com/transmission/transmission/issues/6873#issuecomment-2313508165:

To avoid this kind of problem in the future, would it make sense to have a GitHub Actions workflow to prepare a release?

For example, when pushing a tag that matches a release tag regex, the GitHub Actions workflow could prepare the release assets, and then use something like [`action-gh-release`](https://github.com/softprops/action-gh-release) to create a draft release (without publishing it) with the assets already pre-attached. [Example here](https://github.com/dechamps/FlexASIO/blob/e75cc72483a4f0a4cb906bda80463b0c4bd6e8d3/.github/workflows/continuous-integration.yml#L105-L124). All a human has to do then is to fill out the release description and then click publish. This reduces the potential for human error down to practically zero.

As a bonus, one could also add a workflow step to generate [SLSA provenance](https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md), thereby guaranteeing the authenticity of release artifacts and making Transmission [SLSA Build Level 3](https://slsa.dev/spec/v1.0/levels#build-l3-hardened-builds) compliant.

### Which application of Transmission?

None

**Comment by Pentaphon** [comment]
Any update on this? Do we have an admin who is knowledgeable about implementing this process? I'd like to see this happen from now on because right now, lots of Linux users are stuck on 4.0.5 because Flathub won't build, etc.  Guaranteeing the authenticity of release artifacts sounds quite necessary as well.

**Comment by ckerr** [comment]
No, right now when we need to release, @livings124 manually rolls the macOS binaries, @mikedld manually rolls the Windows releases, and I manually tag the release, make the release notes, and roll the tarball. It's about as inefficient as it gets.

For my own part, I don't feel strongly about any particular approach. I have small concerns about the current method's bus factor. But mainly I just **hate** rolling releases and am happy to leave it to @livings124 and @mikedld to do it however they want. If they want to automate it, fine. If they don't want to automate it, also fine.

Assigning to @livings124 and @mikedld to decide. My assigning y'all does not indicate an endorsement of any approach 😄
